clippy_lints/src/trivial_default_constructed_types.rs

```diff
@@ -0,0 +1,120 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, is_from_proc_macro, is_lang_item_or_ctor, last_path_segment};
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, LangItem, QPath};
+use rustc_lint::{LateContext, LateLintPass, LintContext};
+use rustc_middle::{
+    lint::in_external_macro,
+    ty::{self, Ty},
+};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::kw;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for types constructed by `default` that really don't need to be.
+    ///
+    /// ### Why is this bad?
+    /// It's harder for the reader to know what the value is, and it's an unnecessary function call.
+    ///
+    /// ### Example
+    /// ```rust,ignore
+    /// let a = A(Option::default());
+    /// ```
+    /// Use instead:
+    /// ```rust,ignore
+    /// let a = A(None);
+    /// ```
+    #[clippy::version = "1.72.0"]
+    pub TRIVIAL_DEFAULT_CONSTRUCTED_TYPES,
+    pedantic,
+    "checks for usage of `Default::default` to construct trivial types"
+}
+declare_lint_pass!(TrivialDefaultConstructedTypes => [TRIVIAL_DEFAULT_CONSTRUCTED_TYPES]);
+
+impl<'tcx> LateLintPass<'tcx> for TrivialDefaultConstructedTypes {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &Expr<'tcx>) {
+        if !in_external_macro(cx.sess(), expr.span)
```

Comment:
It would be better to not lint in any macros unless you want to go through a lot of extra effort. A simple example is a macro expanding to `foo(Default::default())` might call different `foo` functions in each expansion.